src/components/themed/TransactionListTop.tsx

```diff
@@ -90,14 +92,18 @@ export class TransactionListTopComponent extends React.PureComponent<Props, Stat
     const { pluginId } = wallet.currencyInfo
 
     if (SPECIAL_CURRENCY_INFO[pluginId]?.isStakingSupported === true) {
-      makeStakePlugin().then(stakePlugin => {
-        const filteredStatePolicies = stakePlugin.policies.filter(stakePolicy => {
-          return [...stakePolicy.rewardAssets, ...stakePolicy.stakeAssets].some(asset => asset.pluginId === pluginId && asset.currencyCode === currencyCode)
-        })
-        this.setState({ stakePolicies: filteredStatePolicies })
+      makeStakePlugins().then(stakePlugins => {
+        let stakePolicies: StakePolicy[] = []
+        for (const stakePlugin of stakePlugins) {
+          const filteredStatePolicies = stakePlugin.policies.filter(stakePolicy => {
+            return [...stakePolicy.rewardAssets, ...stakePolicy.stakeAssets].some(asset => asset.pluginId === pluginId && asset.currencyCode === currencyCode)
+          })
+          stakePolicies = [...stakePolicies, ...filteredStatePolicies]
+        }
+        this.setState({ stakePolicies, stakePlugins })
```

Comment:
I believe this invocation needs to be react to the change in component props because the walletId and pluginId can change for the scene via the wallet list modal. Perhaps create a new method and call it in this life-cycle event and also in the handler for the modal bridge response.

Reply:
I've always hated how componentDidMount and componentDidUpdate are two separate funcs. I'll create a common routine that both of them call